CMakeLists.txt

```diff
@@ -24,6 +24,11 @@ list(APPEND BUILT_WITH "${CMAKE_C_COMPILER_ID}")
 list(APPEND BUILT_WITH "${CMAKE_CXX_COMPILER_ID}")
 # }
 
+if (MINGW AND RDKAFKA_BUILD_STATIC)
+  # ".dll.a" takes precedence over ".a" by default
```

Comment:
Aren't static libs using a `.lib` extension on Windows?

Reply:
This is how MSYS2 distributes their MinGW-w64 static libraries. It looks like this is for historical reasons, as `.a` is what `ld` is looking for and `.lib` is what `LINK.exe` is looking for, but, at some point, someone added `.lib` support to the MinGW `ld`

Reply:
If the Go toolchain supports the .lib extension I think we should go with that.

Reply:
Well, `CMAKE_FIND_LIBRARY_SUFFIXES` is used by CMake to find 3rd party dependencies, e.g. libssl. These are distributed by MSYS2 as `.a`, so this would have to stay.

As for confluent-kafka-go, we can change the bundled `librdkafka_windows.a` to `librdkafka_windows.lib`. This would be a bit of an oddity though, as `.lib` is only commonly used for MSVC static libraries, i.e. those built with `cl.exe`. If someone tries to use `librdkafka_windows.lib` in an MSVC project, or with `LINK.exe`, it will fail to link. This is probably why the convention is continue using `.a` for static libraries built with MinGW `gcc`. (While MinGW `ld` handles both the GNU `.a` format and the Microsoft `.lib` format, `LINK.exe` still only supports the Microsoft `.lib` format)

Reply:
You had me at oddity, let's go with `.a`.